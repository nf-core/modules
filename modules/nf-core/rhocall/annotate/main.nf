process RHOCALL_ANNOTATE {
    tag "$meta.id"
    label 'process_medium'

    conda (params.enable_conda ? "bioconda::rhocall=0.5.1" : null)
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/rhocall:0.5.1--py39hbf8eff0_0':
        'quay.io/biocontainers/rhocall:0.5.1--py39hbf8eff0_0' }"

    input:
    tuple val(meta), path(vcf), path(tbi)
    tuple val(meta), path(roh)
    path bed

    output:
    tuple val(meta), path("*_rhocall.vcf"), emit: vcf
    path "versions.yml"                   , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args   = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def az_bed = bed ? "-b ${bed}" : ''
    """
    rhocall \\
        annotate \\
        $args \\
        $az_bed \\
        -r $roh \\
        -o ${prefix}_rhocall.vcf \\
        $vcf

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        rhocall: \$(echo \$(rhocall --version 2>&1) | sed 's/rhocall, version //' )
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_rhocall.vcf

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        rhocall: \$(echo \$(rhocall --version 2>&1) | sed 's/rhocall, version //' )
    END_VERSIONS
    """
}
