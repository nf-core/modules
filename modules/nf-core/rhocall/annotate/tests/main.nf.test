nextflow_process {

    name "Test Process RHOCALL_ANNOTATE"

    script "../main.nf"
    process "RHOCALL_ANNOTATE"

    tag "modules"
    tag "modules_nfcore"
    tag "rhocall"
    tag "rhocall/annotate"
    tag "bcftools"
    tag "bcftools/roh"

    setup {
        run("BCFTOOLS_ROH") {
            script "../../../bcftools/roh/main.nf"

            process {
                """
                input[0] = [[id:'test'],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test.vcf.gz', checkIfExists:true),
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test.vcf.gz.tbi', checkIfExists:true)
                    ]
                input[1] = [[],[]]
                input[2] = []
                input[3] = []
                input[4] = []
                input[5] = []
                """
            }
        }
    }

    test("sarscov2") {

        when {
            params {
                module_args = ''
            }
            process {
                """
                input[0] = [[id:'test'],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test.vcf.gz', checkIfExists:true),
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test.vcf.gz.tbi', checkIfExists:true)
                    ]
                input[1] = BCFTOOLS_ROH.out.roh
                input[2] = []
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    path(process.out.vcf.get(0).get(1)).vcf.variantsMD5,
                    process.out.versions,
                    path(process.out.versions[0]).yaml
                    ).match() }
            )
        }
    }

    test("sarscov2 - stub") {
        options '-stub'
        when {
            params {
                module_args = ''
            }
            process {
                """
                input[0] = [[id:'test'],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test.vcf.gz', checkIfExists:true),
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test.vcf.gz.tbi', checkIfExists:true)]
                input[1] = BCFTOOLS_ROH.out.roh
                input[2] = []
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

}