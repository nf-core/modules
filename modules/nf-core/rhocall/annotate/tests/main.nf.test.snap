{
    "sarscov2 - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_rhocall.vcf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,9df04f5e1e5975d4b24735cb18806987"
                ],
                "vcf": [
                    [
                        {
                            "id": "test"
                        },
                        "test_rhocall.vcf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,9df04f5e1e5975d4b24735cb18806987"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-07T11:35:38.640805062"
    },
    "sarscov2": {
        "content": [
            "bc7bf3ee9e8430e064c539eb81e59bf9",
            [
                "versions.yml:md5,9df04f5e1e5975d4b24735cb18806987"
            ],
            {
                "RHOCALL_ANNOTATE": {
                    "rhocall": "0.5.1"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-07T11:13:58.507032304"
    }
}